src/TypeSpec/__Private/EnumSpec.hack

```diff
@@ -32,7 +33,24 @@ final class EnumSpec<T as arraykey> extends TypeSpec<T> {
   public function assertType(mixed $value): T {
     $what = $this->what;
     try {
-      return $what::assert($value);
+      $t = $what::assert($value);
+      if (!C\contains($what::getValues(), $value)) {
```

Comment:
Footnote, this is a linear scan. We can't use `C\contains_key($what::getNames(), $value)`, because this throws for enums with duplicate members.